### PR TITLE
Added support for doing a permanent redirect to the PrerenderTagHelper

### PR DESCRIPTION
**Pull request by basschuitema**
Fix for issue https://github.com/aspnet/AspNetCore/issues/6477

### CODE REVIEW

File: src/Middleware/SpaServices/src/Prerendering/PrerenderTagHelper.cs [hunk 1]
```diff
@@ -101,8 +101,10 @@ public override async Task ProcessAsync(TagHelperContext context, TagHelperOutpu
 
             if (!string.IsNullOrEmpty(result.RedirectUrl))
             {
+                var permanentRedirect = result?.StatusCode == 301;
```
Comment (SteveSandersonMS) [inline src/Middleware/SpaServices/src/Prerendering/PrerenderTagHelper.cs]: No need for the `?` in this line. We already know from L102 that `result` is nonnull.

**Review by SteveSandersonMS — COMMENTED** [review]
Looks great to me! Apart from the one trivial tweak I've suggested, this looks like an ideal solution.

File: src/Middleware/SpaServices/src/Prerendering/PrerenderTagHelper.cs (hunk 1 above)
Comment (basschuitema) [reply src/Middleware/SpaServices/src/Prerendering/PrerenderTagHelper.cs]: That's true, thanks for looking at it! I updated the PR.

**Review by SteveSandersonMS — APPROVED** [review]

**Comment by ryanbrandenburg** [comment]
@SteveSandersonMS is always doing a permanent redirect the correct behavior here? I don't know one way or another but just thought that I'd point out that this isn't "support" for permanent redirects (as in you can do it either way) but rather permanent only in this case.

**Comment by SteveSandersonMS** [comment]
@ryanbrandenburg I don't follow. The code looks to me like it only sets the `permanentRedirect` flag conditionally based on the `result.StatusCode` that is set by developer code. By default, `result.StatusCode` won't equal 301, so `permanentRedirect` would be `false`.

**Review by ryanbrandenburg — APPROVED** [review]

**Comment by mkArtakMSFT** [comment]
Thanks for your effort, @basschuitema.
Can you please rebase so that we can merge this? Thanks!

**Comment by basschuitema** [comment]
It is already merged to master by another pull request, see: #7179. 
I will close this pull request.

**Pull request closed without merging**